### PR TITLE
net-misc/oidentd: add 3.0.0

### PR DESCRIPTION
**Pull request by Dessa**

### CODE REVIEW

File: net-misc/oidentd/oidentd-3.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+DEPEND="masquerade? ( net-libs/libnetfilter_conntrack )"
+
+RDEPEND="
+	acct-user/oidentd
+	acct-group/oidentd
+	selinux? ( sec-policy/selinux-oident )
+	${DEPEND}"
+
+src_prepare() {
+	sed -i '/ExecStart/ s|$| -u oidentd -g oidentd|' contrib/systemd/*.service || die
```
Comment (juippis) [inline net-misc/oidentd/oidentd-3.0.0.ebuild:26]:
```
 * QA: following sed did not cause any changes
 *     sed -i '/ExecStart/ s|$| -u oidentd -g oidentd|' contrib/systemd/*.service || die;
```

File: net-misc/oidentd/oidentd-3.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+IUSE="debug ipv6 masquerade selinux"
+
+DEPEND="masquerade? ( net-libs/libnetfilter_conntrack )"
+
+RDEPEND="
+	acct-user/oidentd
+	acct-group/oidentd
+	selinux? ( sec-policy/selinux-oident )
+	${DEPEND}"
+
```
Comment (juippis) [inline net-misc/oidentd/oidentd-3.0.0.ebuild:24]: This requires sys-devel/flex and sys-devel/bison as BDEPEND.

**Review by juippis — COMMENTED** [review]
```
[ERROR] >>> Not all runs were successful.
atom: =net-misc/oidentd-3.0.0, USE flags: '-ipv6 masquerade'
```
```
os.c:524:25: warning: implicit declaration of function 'sin_setv6'; did you mean 'sin_setv4'? [-Wimplicit-function-declaration[https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration]]
  524 |                         sin_setv6(&localm6, &localm_ss);
      |                         ^~~~~~~~~
      |                         sin_setv4
x86_64-pc-linux-gnu-gcc   -march=native -O2 -pipe -frecord-gcc-switches  -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -o oidentd oidentd.o util.o inet_util.o forward.o user_db.o options.o masq.o cfg_scan.o cfg_parse.o os.o -Lmissing -lmissing  -lnetfilter_conntrack 
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: os.o: in function `masq_ct_line.part.0':
os.c:(.text+0x7ec): undefined reference to `sin_setv6'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: os.c:(.text+0x80e): undefined reference to `sin_setv6'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: os.c:(.text+0x81d): undefined reference to `sin_setv6'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: os.c:(.text+0x82c): undefined reference to `sin_setv6'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: os.c:(.text+0x9ce): undefined reference to `get_user6'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:462: oidentd] Error 1
make[4]: Leaving directory '/var/tmp/portage/net-misc/oidentd-3.0.0/work/oidentd-3.0.0/src'
make[3]: *** [Makefile:514: all-recursive] Error 1
make[3]: Leaving directory '/var/tmp/portage/net-misc/oidentd-3.0.0/work/oidentd-3.0.0/src'
make[2]: *** [Makefile:381: all] Error 2
make[2]: Leaving directory '/var/tmp/portage/net-misc/oidentd-3.0.0/work/oidentd-3.0.0/src'
make[1]: *** [Makefile:433: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/net-misc/oidentd-3.0.0/work/oidentd-3.0.0'
make: *** [Makefile:353: all] Error 2
 * ERROR: net-misc/oidentd-3.0.0::gentoo failed (compile phase):
 *   emake failed
```

@conikost could you look to the issues I reported?

File: net-misc/oidentd/oidentd-3.0.0.ebuild (hunk 1 above)
Comment (ConiKost) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: He, why I didn't get that QA warning during compilation?

Comment (thesamesam) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: Custom tool: https://github.com/ionenwks/iwdevtools

Comment (juippis) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: It's from iwdevtools - https://packages.gentoo.org/packages/app-portage/iwdevtools still betatesting this though, but if it gives such an error, it's nice to ask for a confirmation whether it does anything or not. 

Now I can see the asterisk in here either not matching anything, or it getting missed by ionen's qa-sed.

Comment (ConiKost) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: Good to know, thanks!

Comment (ConiKost) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: I think, this warning is a false positive? After installing `iwdevtools` I am getting this warning, but if I do run manally the command, only without `|| die `, all *.service files have been changed correctly?

Comment (ConiKost) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:26]: All files in /var/tmp/portage are also correctly.

File: net-misc/oidentd/oidentd-3.0.0.ebuild (hunk 2 above)
Comment (ConiKost) [reply net-misc/oidentd/oidentd-3.0.0.ebuild:24]: Added them. Good catch, thank you.

**Pull request closed without merging**